### DISCUSSION TITLE
24.08

### DISCUSSION THREAD

**Discussion by myfreeer** [myfreeer/7z-build-nsis#56] [Announcements]
7-zip 24.08 with NSIS script decompiling
* `7z****-x64.exe`: 7-Zip for 64-bit Windows (x64).
* `7z****-x86.exe`: 7-Zip for 32-bit Windows (x86).
* `7z****.7z`: Everything compiled.

Compatibility
-------------------------
* 7-Zip for 64-bit Windows (x64): Windows Vista and above.
* 7-Zip for 32-bit Windows (x86): Windows XP (maybe, not tested) and above.

24.08          2024-08-11
-------------------------
- The bug in 7-Zip 24.00-24.07 was fixed:
  For creating a zip archive: 7-Zip could write extra zero bytes after the end of the archive,
  if a file included to archive cannot be compressed to a size smaller than original.
  The created zip archive is correct except for the useless zero bytes after the end of the archive.
  When unpacking such a zip archive, 7-Zip displays a warning:
    "WARNING: There are data after the end of archive".
- The bug was fixed: there was a leak of GDI objects (internal resources in Windows)
  in "Confirm File Replace" window, causing problems after 1600 displays of "Confirm File Replace"
  window from same running 7-Zip process.
- Some optimizations for displaying file icons in 7-Zip File Manager and in "Confirm File Replace" window.
- Some bugs were fixed.

<hr /><em>This discussion was created from the release <a href='https://github.com/myfreeer/7z-build-nsis/releases/tag/24.08'>24.08</a>.</em>

**Comment by grouik1er-coder** [comment]
hello, is it possible to modify the script to add cflags for an i7-2700 or an atom N2600.
MMX up to AVX in 64 bits and MMX up to SSSE3 and 32 bits to test if result is faster than standard build with buildin 7z benchmark ?

**Reply by myfreeer** [reply]
Check `CPP/Build.mak` file, CFLAGS can be found there

**Comment by grouik1er-coder** [comment]
Thanks,

i try to modifiy line 72 with CFLAGS = $(CFLAGS) -nologo -c -Fo$O/ $(CFLAGS_WARN_LEVEL) -WX -EHsc -Gy -GR- -GF /arch:AVX2 /fp:fast /Oi /MT  but no noticeable speed changes, i don't know if we can go any further.

i will give a try for a 32 bit build for atom.
